### ISSUE TITLE
Create `Adopting.md`

### ISSUE THREAD

**Issue by tsmaeder** [eclipse-theia/theia#14589]
### Feature Description:
There are cases like [#14586](https://github.com/eclipse-theia/theia/issues/14586) where we cannot fix a problem for adopters in our own code. It would be good if we had some documentation for adopters on what they need to do when creating their own application. There is some documentation on theia-ide.org, but it's hard to keep that in sync with the actual code.

**Comment by vince-fugnitto** [comment]
@tsmaeder note that https://github.com/eclipse-theia/theia/blob/master/doc/Migration.md exists, but perhaps it's clearer with the name you proposed and at the root of the repository.

**Comment by tsmaeder** [comment]
Yes, that document is aimed at folks who adopt new releases, but we have no guidance in-repo about how to start a new Theia-based product.

**Comment by tortmayr** [comment]
👍🏼 In my opinion a in-repo documentation for adopters would be a great addition.
Another case that should be mentioned there is a potential build error related to the inversify version.
This was fixed for Theia >=1.56 with https://github.com/eclipse-theia/theia/pull/14435
However, if adopters start from scratch with an older version they need to manually lock (i.e. via `resolutions`) a compatible version.
